mpc-core/src/protocols/rep3/conversion.rs

```diff
@@ -159,3 +166,229 @@ pub fn bit_inject<F: PrimeField, N: Rep3Network>(
+    io_context: &mut IoContext<N>,
+) -> IoResult<Rep3BigUintShare<F>> {
+    let bitlen = x.size();
+    let collapsed = GCUtils::collapse_bundle_to_lsb_bits_as_biguint(x);
+
+    let converted = match io_context.id {
+        PartyID::ID0 => {
+            let x_xor_px = collapsed;
+            let r = io_context.rngs.rand.random_biguint_rng1(bitlen);
+            let r_xor_x_xor_px = x_xor_px ^ &r;
+            io_context
+                .network
+                .send(PartyID::ID2, r_xor_x_xor_px.to_owned())?;
+            Rep3BigUintShare::new(r_xor_x_xor_px, r)
+        }
+        PartyID::ID1 => {
+            let px = collapsed;
+            let r = io_context.rngs.rand.random_biguint_rng2(bitlen);
+            Rep3BigUintShare::new(r, px)
+        }
+        PartyID::ID2 => {
+            let px = collapsed;
+            let r_xor_x_xor_px = io_context.network.recv(PartyID::ID0)?;
+            Rep3BigUintShare::new(px, r_xor_x_xor_px)
+        }
+    };
+
+    Ok(converted)
+}
+
+/// Transforms the replicated shared value x from an arithmetic sharing to a binary sharing. I.e., x = x_1 + x_2 + x_3 gets transformed into x = x'_1 xor x'_2 xor x'_3. .
+pub fn a2y2b<F: PrimeField, N: Rep3Network, R: Rng + CryptoRng>(
+    x: Rep3PrimeFieldShare<F>,
+    io_context: &mut IoContext<N>,
+    rng: &mut R,
+) -> IoResult<Rep3BigUintShare<F>> {
+    let delta = io_context.rngs.generate_random_garbler_delta(io_context.id);
+    let y = a2y(x, delta, io_context, rng)?;
+    y2b(y, io_context)
+}
+
+/// Transforms the replicated shared value x from a binary sharing to an arithmetic sharing. I.e., x = x_1 xor x_2 xor x_3 gets transformed into x = x'_1 + x'_2 + x'_3. This implementations goes through the yao protocol and currently works only for a binary sharing of a valid field element, i.e., x = x_1 xor x_2 xor x_3 < p.
+///
+/// Keep in mind: Only works if the input is actually a binary sharing of a valid field element
+/// If the input has the correct number of bits, but is >= P, then either x can be reduced with self.low_depth_sub_p_cmux(x) first, or self.low_depth_binary_add_2_mod_p(x, y) is extended to subtract 2P in parallel as well. The second solution requires another multiplexer in the end.
+pub fn b2y2a<F: PrimeField, N: Rep3Network, R: Rng + CryptoRng>(
+    x: &Rep3BigUintShare<F>,
+    io_context: &mut IoContext<N>,
+    rng: &mut R,
+) -> IoResult<Rep3PrimeFieldShare<F>> {
+    let delta = io_context.rngs.generate_random_garbler_delta(io_context.id);
+    let y = b2y(x, delta, io_context, rng)?;
+    y2a(y, delta, io_context, rng)
+}
+
+/// Transforms the replicated shared value x from a binary sharing to an arithmetic sharing. I.e., x = x_1 xor x_2 xor x_3 gets transformed into x = x'_1 + x'_2 + x'_3. his implementations goes through the yao protocol and currently works only for a binary sharing of a valid field element, i.e., x = x_1 xor x_2 xor x_3 < p.
+pub fn b2y2a_consume<F: PrimeField, N: Rep3Network, R: Rng + CryptoRng>(
```

Comment:
do we need the consume variants?

Reply:
I don't know, the old b2a has it as well. Not sure who added it though...

Reply:
have removed both